content/docs/support-and-ops/ops-recipes/troubleshooting-gitops.md

```diff
@@ -0,0 +1,69 @@
+---
+title: "Troubleshooting GitOps"
+owner:
+- https://github.com/orgs/giantswarm/teams/team-honeybadger
+confidentiality: public
+---
+
+Since we are offering GitOps as interface for our customers, here we collect to tips on how to troubleshoot the problems can occur. 
+
+# Table of Contents
+1. [Identify which kustomization owns a resource](#identify-which-kustomization-owns-a-resource)
+2. [Download the Git Repository source](#Download-the-Git-Repository-source)
+3. [Stop GitOps reconciliation](#Stop-GitOps-reconciliation)
+
+
+## Identify which kustomization owns a resource
+
+1) Usually the resources contain a set of labels that identify which kustomization belongs to. Example:
+
+```
+  labels:
+    ...
+    kustomize.toolkit.fluxcd.io/name: gorilla-clusters-rfjh2
+    kustomize.toolkit.fluxcd.io/namespace: default
+```
+
+From the kustomization you can the Git Repository looking at the spec field `sourceRef`.
+
+2) Use flux command line. It offers a subcommand `trace` that describe all details related to GitOps:
+
+```
+» flux trace app/alfred-app -n 
+
+Object:        App/alfred-app
+Namespace:     rfjh2
+Status:        Managed by Flux
+---
+Kustomization: gorilla-clusters-rfjh2
+Namespace:     default
+...
+---
+GitRepository: workload-clusters-fleet
+Namespace:     default
+...
+```
+
+__Note__: If the resource has not labels (or command returns `object not managed by Flux`) the object is not produce as result of helm or kustomize but it could be a owner by a higher resource. Like a pod which does not have the labels but the parent deployment can have them.
+
+## Download the Git Repository source
+
+Giant Swarm engineers usually have no access to the customer repository but in case of emergency they need to verify something in the customer source repository. For that matters, they can download the repository temporaly with:
+
+```sh
+export SC=$(kubectl get po -n flux-system -l app=source-controller -o custom-columns=NAME:.metadata.name --no-headers)
+export GITREPO_NAME=<GOT_IT_FROM_PREVIOUS_STEP>
+kubectl cp -n flux-system $SC:/data/gitrepository/default/$GITREPO_NAME/ .
```

Comment:
I tried this command with `gorilla/rfjh2` and got the following error:
```
tar: /data/gitrepository/default/workfload-clusters-fleet: No such file or directory
tar: error exit delayed from previous errors
```

However, just running `kubectl cp -n flux-system $SC:/data/gitrepository/default/ .` created the expected `<COMMIT_SHA>.tar.gz`.

Reply:
but if you dont add repo name it copies all repos. I tested in `gorilla` and it worked for me

```
export GITREPO_NAME=workload-clusters-fleet                                                                                                     
----------------------------------------------------------------------------------------------
» kubectl cp -n flux-system $SC:/data/gitrepository/default/$GITREPO_NAME/ .  
tar: removing leading '/' from member names
warning: skipping symlink: "latest.tar.gz" -> "/data/gitrepository/default/workload-clusters-fleet/87dce26c243e947720ed67641aeb98e4701c56e3.tar.gz" (consider using "kubectl exec -n "" "source-controller-f4d9c4dfb-qk2pw" -- tar cf - "/data/gitrepository/default/workload-clusters-fleet" | tar xf -")
-------------------------------------------------------------------
» ls             
87dce26c243e947720ed67641aeb98e4701c56e3.tar.gz
```

Reply:
I tested again and it works, you are right.

Maybe I was logged into a workload cluster on my first try?

Whatever, *your* version is correct.